### PR TITLE
Change naming and packaging.

### PR DESCRIPTION
**Pull request by ming13**
* No more formatters, only renderers.
* Moved templates and classes to `feed` and `website` sub-packages and sub-directories.
* Renamed `MarkdownRenderer` to `HtmlRenderer` since it actually makes sense.
* Renamed `MustacheRenderer` to `TemplateRenderer` since ideally we shouldn’t care about which template language we are using.

### CODE REVIEW

File: src/main/kotlin/io/thecontext/ci/output/TemplateRenderer.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        WebsiteEpisode("website/episode.md.mustache"),
+    }
+
+    fun render(template: Template, contents: Map<String, Any>): String
+
+    class Impl : TemplateRenderer {
+
+        private val factory by lazy { DefaultMustacheFactory() }
+
+        override fun render(template: Template, contents: Map<String, Any>): String {
```
Comment (sockeqwe) [inline src/main/kotlin/io/thecontext/ci/output/TemplateRenderer.kt]: NIT: If we talk about independent Template engines, then imho parameter `contents` should not be `Map<String, Any>` but rather of type Episode / prepared episode.

File: src/main/kotlin/io/thecontext/ci/output/HtmlRenderer.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package io.thecontext.ci.output
+
+import org.commonmark.parser.Parser
+
+interface HtmlRenderer {
+
+    fun render(markdown: String): String
```
Comment (sockeqwe) [inline src/main/kotlin/io/thecontext/ci/output/HtmlRenderer.kt]: NIT: why gets a HtmlRendere markdown as a String? Isn't it an implementation detail and should be of type episode (and transform it under the hood to markdown then to html)

File: src/main/kotlin/io/thecontext/ci/output/TemplateRenderer.kt (hunk 1 above)
Comment (ming13) [reply src/main/kotlin/io/thecontext/ci/output/TemplateRenderer.kt]: Even better would be receiving a `data class` abstracting from our own datatypes...

File: src/main/kotlin/io/thecontext/ci/output/HtmlRenderer.kt (hunk 2 above)
Comment (ming13) [reply src/main/kotlin/io/thecontext/ci/output/HtmlRenderer.kt]: This way `HtmlRenderer` doesn’t know about outside world types. Single responsibility, basically.

**Pull request merged**